include/os/freebsd/spl/sys/uio.h

```diff
@@ -39,31 +39,43 @@
 
 #define	uio_loffset	uio_offset
 
-typedef	struct uio	uio_t;
 typedef	struct iovec	iovec_t;
 typedef	enum uio_seg	uio_seg_t;
 
+typedef struct zfs_uio {
+	struct uio	*uio;
+} uio_t;
```

Comment:
We can head off any potential future header conflict now by defining this as a `zfs_uio_t` instead.  Renaming the `uio_*` wrapper functions to `zfs_uio_*` would also probably be in order.  That should probably even include `uiocopy`, `uiomove`, `uioskip`.  That would make it clear this is now really a zfs interface and not just a compatibility for the original illumos interface.

It looks like there are about 71 places where a `uio_t` would need to be changed to a `zfs_uio_t`.  It'd be a slightly bigger change but I'd be fine with that.  It looks like you're already updating most of them.

Reply:
I went ahead and updated all the interfaces to now be zfs_uio_*.

Reply:
\o/